libs/partners/pinecone/langchain_pinecone/embeddings.py

```diff
@@ -0,0 +1,154 @@
+import logging
+import os
+from typing import Dict, Iterable, List, Optional
+
+import aiohttp
+from langchain_core.embeddings import Embeddings
+from langchain_core.pydantic_v1 import (
+    BaseModel,
+    Extra,
+    Field,
+    SecretStr,
+    root_validator,
+)
+from langchain_core.utils import convert_to_secret_str
+from pinecone import Pinecone as PineconeClient  # type: ignore
+
+logger = logging.getLogger(__name__)
+
+
+class PineconeEmbeddings(BaseModel, Embeddings):
+    """PineconeEmbeddings embedding model.
+
+    Example:
+        .. code-block:: python
+
+            from langchain_pinecone import PineconeEmbeddings
+
+            model = PineconeEmbeddings()
+    """
+
+    _client: PineconeClient = Field(exclude=True)
+    _async_client: aiohttp.ClientSession = Field(exclude=True)
+    model: str
+    batch_size: int
+    show_progress_bar: bool = False
+    truncation: Optional[bool] = None
+    pinecone_api_key: Optional[SecretStr] = None
+
+    class Config:
+        extra = Extra.forbid
+
+    @root_validator(pre=True)
+    def default_values(cls, values: dict) -> dict:
+        """Set default batch size"""
+
+        batch_size = values.get("batch_size")
+        if batch_size is None:
+            values["batch_size"] = 128
+        return values
+
+    @root_validator()
+    def validate_environment(cls, values: dict) -> dict:
+        """Validate that Pinecone credentials exist in environment."""
+        pinecone_api_key = values.get("pinecone_api_key") or os.getenv(
+            "PINECONE_API_KEY", None
+        )
+        if pinecone_api_key:
+            api_key_secretstr = convert_to_secret_str(pinecone_api_key)
+            values["pinecone_api_key"] = api_key_secretstr
+
+            api_key_str = api_key_secretstr.get_secret_value()
+        else:
+            api_key_str = None
+        values["_client"] = PineconeClient(api_key=api_key_str, source_tag="langchain")
```

Comment:
I'm assuming that the PineconeClient will throw an error if the api_key is None, which is why error handling is not done here, right?

Reply:
it will, but actually we should have consistent error messages between the sync and async client initialization so will more the `ValueError` to above this line

Reply:
^ done